scripts/zkai_native_d128_verifier_execution_compression_budget_gate.py

```
@@ -0,0 +1,679 @@
+
+def validate_output_path(path: pathlib.Path) -> pathlib.Path:
+    raw_path = path if path.is_absolute() else ROOT / path
+    reject_symlinked_path_components(EVIDENCE_DIR, "evidence dir")
+    reject_symlinked_path_components(raw_path, "output path")
+    resolved = raw_path.resolve()
+    try:
+        resolved.relative_to(EVIDENCE_DIR.resolve())
+    except ValueError as err:
+        raise CompressionBudgetGateError(f"output path must be under evidence dir: {raw_path}") from err
+    if not resolved.parent.exists():
+        raise CompressionBudgetGateError(f"output parent does not exist: {resolved.parent}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)